phpBB/phpbb/db/migration/data/v33x/bot_update.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+namespace phpbb\db\migration\data\v33x;
+
+class bot_update extends \phpbb\db\migration\migration
+{
+	static public function depends_on()
+	{
+		return ['\phpbb\db\migration\data\v330\v330'];
+	}
+
+	public function update_data()
+	{
+		return [
+			['custom', [[$this, 'add_duckduckgo_bot']]],
+		];
+	}
+
+	public function add_duckduckgo_bot()
+	{
+		$bot_name = 'DuckDuckGo [Bot]';
+		$bot_name_clean = utf8_clean_string($bot_name);
+
+		$sql = 'SELECT user_id
```

Comment:
Question: Does it make sense ot check for $bot_agent in the "bot_exists" SQL-query. Because when some user added it himself, the name can be more or less anything, but not he user agent for the same bot.

Reply:
@Crizz0 Totally, however `bot_agent` is not guaranteed to be unique in the table, you can have two bots with the same `bot_agent` but different username.

You cannot, however, have two users with the same name, hence the check in the users table.

I could check for bots with that `bot_agent` and update its bot and user data if it already exists, in case a user tried to impersonate a bot, however what should I do if there's more than one? delete other to leave only one and edit that one?

Reply:
I have Duck there as `DuckDuckBot/1.0; (+http://duckduckgo.com/duckduckbot.html)`

![immagine](https://user-images.githubusercontent.com/480857/78711871-ffbce680-7917-11ea-9fa5-170422a7259d.png)

Reply:
@3D-I Yeah, that's another problem, bots with different `bot_agent`, however that would be a little easier to "fix" with a `preg_match()`, because they both start with `DuckDuckBot/`.

In my opinion, the `bot_agent` should be a partial match, because the version (for example) can change overtime. However that partial match should be specific enough to allow to identify that and only that bot.

In fact there's a `DuckDuckBot/1.1` user agent ([reference])

[reference]: https://github.com/duckduckgo/duckduckgo-publisher/blob/master/share/site/duckduckgo/duckduckbot.tx

Reply:
Yeah, your proposal is right as it is IMO. :+1: